lib/virt_autotest/utils.pm

```diff
@@ -85,19 +104,45 @@ sub restart_modular_libvirt_daemons {
     } else {
         # Restart the sockets first
         foreach my $daemon (@daemons) {
-            systemctl("restart $daemon\{,-ro,-admin\}.socket");
+            if ($daemon == 'lock') {
+                systemctl("restart virt${daemon}d\{,-admin\}.socket");
+            } else {
+                systemctl("restart virt${daemon}d\{,-ro,-admin\}.socket");
+            }
         }
 
         # Introduce idle time here (e.g., sleep 5) if necessary
         sleep 5;
 
         # Restart the services after a brief idle time
         foreach my $daemon (@daemons) {
-            systemctl("restart $daemon.service");
+            systemctl("restart virt${daemon}d.service");
         }
     }
 
-    record_info("Libvirt daemons restarted", join(' ', @daemons));
+    record_info("Modular Libvirt daemons restarted", join(' ', @daemons));
+    check_modular_libvirt_daemons;
+}
+
+# Restart monolithic libvirt
+sub restart_libvirtd {
+    if (is_sle('<12')) {
+        assert_script_run('rclibvirtd restart', 180);
+    }
+    elsif (is_alp) {
+        my $_libvirtd_pid = script_output(q@ps -ef |grep [l]ibvirtd | gawk '{print $2;}'@);
+        my $_libvirtd_cmd = script_output("ps -o command $_libvirtd_pid | tail -1");
+        assert_script_run("kill -9 $_libvirtd_pid");
+        assert_script_run("$_libvirtd_cmd");
+    }
+    elsif (is_modular_libvirtd) {
+        restart_modular_libvirt_daemons;
```

Comment:
if `restart_modular_libvirt_daemons` is called inside `restart_libvirtd`, there are more places need to be adapted to this change. fg. `is_monolithic_libvirtd ? restart_libvirtd : restart_modular_libvirt_daemons;` this kind of statements.

Reply:
Search "libvirtd" in `lib`, `tests/virt_autotest` and `tests/virtualization`, you can get more.

Reply:
Super! make sense now. will do more validation tests with them.

Reply:
> if `restart_modular_libvirt_daemons` is called inside `restart_libvirtd`, there are more places need to be adapted to this change. fg. `is_monolithic_libvirtd ? restart_libvirtd : restart_modular_libvirt_daemons;` this kind of statements.

@Julie-CAO This is expected. IMHO, wrapping `if modular ... else ...` in one function, will save the if else in all places that call it, so will make code simpler after modular libvirt task is fully done. Besides, such statements are limited, just several, right? So overall, it is a preferred solution for me. How about you? Do I overlook any important factor when judging this?

Reply:
I am ok with this logic, but existing codes are required to be adapted as I said above.

Reply:
> Search "libvirtd" in `lib`, `tests/virt_autotest` and `tests/virtualization`, you can get more.

Indeed, try to search both libvirt and libvirtd from `lib`, `tests/virt_autotest` and `tests/virtualization`. 
hope that current code changes would cover that adapt modular libvirt daemons.